cinema/views.py

```diff
@@ -43,6 +55,30 @@ def get_serializer_class(self):
 
         return MovieSerializer
 
+    def get_queryset(self):
+        genres = self.request.query_params.get("genres")
+        actors = self.request.query_params.get("actors")
+        title = self.request.query_params.get("title")
+
+        if title:
+            self.queryset = self.queryset.filter(title__icontains=title)
+
+        if actors:
+            actors = [int(actor) for actor in actors.split(",")]
+            self.queryset = self.queryset.filter(actors__id__in=actors)
+        if genres:
+            genres = [int(genre) for genre in genres.split(",")]
```

Comment:
it's better to create a function in order to follow DRY
also no need to use id for many to many, just actors__in should be enough